common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -488,6 +543,28 @@ public UTF8String trimLeft() {
     }
   }
 
+  /**
+   * Removes all specified trim character from the beginning of a string
+   * @param trimChar the trim character
+   */
+  public UTF8String trimLeft(UTF8String trimChar) {
+    int numTrimBytes = trimChar.numBytes;
+    if (numTrimBytes == 0) {
+      return this;
+    }
+    int s = 0;
+    // skip all the consecutive matching character in the left side
+    while(s < this.numBytes && s == this.find(trimChar, s)) {
```

Comment:
`this.find(trimChar, s)` in a loop probably causes performance issue. I will suggest to re-implement it other than call the existing function.